### ISSUE TITLE
[Accessibility] Text widget wrapped with Semantics(header: true) does not announce as a heading on Android

### ISSUE THREAD

**Issue by iamwalter** [flutter/flutter#179340]
### Steps to reproduce

Hello! 

Since Flutter 3.38.1, Text widgets do not get announced as a 'heading' when surrounded by a `Semantics(heading: true)`. This is only happening on Android.

### Expected results

When gaining focus on 'This is my header' the text gets read aloud and there is an announcement that it is a heading.

### Actual results

When gaining accessibility focus, the text is just being read aloud without announcing that this is a header.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(const HeaderSemanticsBugApp());
}

class HeaderSemanticsBugApp extends StatelessWidget {
  const HeaderSemanticsBugApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Semantics Test App'),
        ),
        body: Center(
          child: Semantics(
            header: true, // <-- Android seems to ignore this on Flutter >=3.38
            child: const Text("This is my header"),
          ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>
Here is a video of it working in Flutter 3.32 (also worked in flutter 3.35):

https://github.com/user-attachments/assets/794eb2df-3144-4989-8dc9-92f164146b29

It seems to have broken in Flutter 3.38:

https://github.com/user-attachments/assets/4596d672-1742-465a-8a14-f642f342d748
</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel [user-branch], 3.38.3, on macOS 15.7.2 24G325 darwin-arm64,
    locale en-NL) [415ms]
    ! Flutter version 3.38.3 on channel [user-branch] at
      /Users/walter/dev/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an
      official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions
      at https://flutter.dev/setup.
    ! Upstream repository unknown source is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss
      this error.
    • Framework revision 19074d12f7 (12 days ago), 2025-11-20 17:53:13 -0500
    • Engine revision 13e658725d
    • Dart version 3.10.1
    • DevTools version 2.51.1
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop,
      enable-windows-desktop, enable-android, enable-ios, cli-animations,
      enable-native-assets, omit-legacy-version-file, enable-lldb-debugging
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    [1,261ms]
    • Android SDK at /Users/walter/Library/Android/sdk
    • Emulator version 36.1.9.0 (build_id 13823996) (CL:N/A)
    • Platform android-36, build-tools 35.0.0
    • Java binary at: /opt/homebrew/opt/openjdk@17/bin/java
      This JDK is specified in your Flutter configuration.
      To change the current JDK, run: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment Homebrew (build 17.0.15+0)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor
      --android-licenses

[!] Xcode - develop for iOS and macOS (Xcode 26.1.1) [1,123ms]
    • Xcode at /Applications/Xcode_nieuw.app/Contents/Developer
    • Build 17B100
    ! CocoaPods 1.15.2 out of date (1.16.2 is recommended).
        CocoaPods is a package manager for iOS or macOS platform code.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/to/platform-plugins
      To update CocoaPods, see
      https://guides.cocoapods.org/using/getting-started.html#updating-cocoapods

[✓] Chrome - develop for the web [6ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (6 available) [7.2s]
    • SM A556B (mobile)           • RZCXC0V0QMV                          • android-arm64  • Android 16 (API 36)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 15 (API 35) (emulator)
    • W (wireless) (mobile)       • 00008110-001C145C1A0BA01E            • ios            • iOS 18.6.2 22G100
    • iPhone 17 (mobile)          • BA011149-8902-467A-ADCD-7E596CC9298A • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-26-1 (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 15.7.2 24G325
      darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 142.0.7444.176

[✓] Network resources [201ms]
    • All expected network resources are available.

! Doctor found issues in 3 categories.

```

</details>

**Comment by iamwalter** [comment]
Seems I did not add 'headingLevel' as a parameter. This works;


```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(const HeaderSemanticsBugApp());
}

class HeaderSemanticsBugApp extends StatelessWidget {
  const HeaderSemanticsBugApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Semantics Test App'),
        ),
        body: Center(
          child: Semantics(
            header: true, // <-- Android seems to ignore this on Flutter >=3.38
            headingLevel: 1,
            child: const Text("This is my header"),
          ),
        ),
      ),
    );
  }
}
```

Hopefully this serves as documentation for someone in the future :). Closing this issue.
